django_usaid_clinicfinder/clinicfinder/tests.py

```diff
@@ -271,4 +271,84 @@ def test_create_lookuppointofinterest_model_data_no_result(self):
+                         'Street Address': ''}
+    CSV_LINE_CLEAN_2 = {'Area 1': 'Pixley ka Seme',
+                         'Area 2': '',
+                         'Area 3': '',
+                         'Clinic Name': 'HCT Clinic',
+                         'HCT': 'true',
+                         'ID': '5633',
+                         'Latitude': '-29.66648',
+                         'Longitude': '22.73116',
+                         'MMC': 'false',
+                         'Primary Contact Number': '',
+                         'Province': 'Northern Cape',
+                         'Street Address': ''}
+    CSV_LINE_DUP_LOC = {'Area 1': 'Pixley ka Seme',
+                         'Area 2': '',
+                         'Area 3': '',
+                         'Clinic Name': 'MMC Clinic',
+                         'HCT': 'false',
+                         'ID': '5631',
+                         'Latitude': '-29.66648',
+                         'Longitude': '22.73116',
+                         'MMC': 'true',
+                         'Primary Contact Number': '538022222',
+                         'Province': 'Northern Cape',
+                         'Street Address': ''}
+
+    @override_settings(CELERY_EAGER_PROPAGATES_EXCEPTIONS = True,
+                       CELERY_ALWAYS_EAGER = True,
+                       BROKER_BACKEND = 'memory',)
+    def setUp(self):
+        self.admin = User.objects.create_superuser(
+            'test', 'test@example.com', "pass123")
+
+    def test_upload_view_not_logged_in_blocked(self):
+        response = self.client.get(reverse("locations_uploader"))
+        # redirect
+        self.assertEqual(response.status_code, 302)
+
+    def test_upload_view_logged_in(self):
+        self.client.login(username="test", password="pass123")
+
+        response = self.client.get(reverse("locations_uploader"))
+        self.assertIn("Upload Locations CSV", response.content)
+
+    def test_upload_csv(self):
+        clean_sample = list()
+        clean_sample.append(self.CSV_LINE_CLEAN_1)
+        clean_sample.append(self.CSV_LINE_CLEAN_2)
+        results = PointOfInterest_Importer.delay(clean_sample)
+        self.assertEqual(results.get(), 2)
+        new_locations = Location.objects.all().count()
+        self.assertEquals(new_locations, 2)
+        new_pois = PointOfInterest.objects.all().count()
+        self.assertEquals(new_pois, 2)
+
+
+    def test_upload_csv_dupe_locations(self):
+        dupe_sample = list()
+        dupe_sample.append(self.CSV_LINE_CLEAN_1)
+        dupe_sample.append(self.CSV_LINE_CLEAN_2)
+        dupe_sample.append(self.CSV_LINE_DUP_LOC)
+        results = PointOfInterest_Importer.delay(dupe_sample)
+        self.assertEqual(results.get(), 3)
+        new_locations = Location.objects.all().count()
+        self.assertEquals(new_locations, 2)
+        new_pois = PointOfInterest.objects.all().count()
+        self.assertEquals(new_pois, 3)
+
```

Comment:
Perhaps worth adding a test here with a CSV_LINE that doesn't contain a Latitude coordinate.